Doppler.Sap/Factory/UpdateBusinessPartnerHandler.cs

```diff
@@ -0,0 +1,132 @@
+using Doppler.Sap.Models;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Doppler.Sap.Factory
+{
+    public class UpdateBusinessPartnerHandler
+    {
+        private readonly SapConfig _sapConfig;
+        private readonly ISapTaskHandler _sapTaskHandler;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public UpdateBusinessPartnerHandler(
+            IOptions<SapConfig> sapConfig,
+            ISapTaskHandler sapTaskHandler,
+            IHttpClientFactory httpClientFactory)
+        {
+            _sapConfig = sapConfig.Value;
+            _sapTaskHandler = sapTaskHandler;
+            _httpClientFactory = httpClientFactory;
+        }
+
+        public async Task<SapTaskResult> Handle(SapTask dequeuedTask)
+        {
+            //SAP uses a non conventional patch where you have to send only the fields that you want to be changed with the new values
+            dequeuedTask.BusinessPartner.BPAddresses = GetBPAddressesPatchObject(dequeuedTask.BusinessPartner.BPAddresses);
+            dequeuedTask.BusinessPartner.ContactEmployees = GetContactEmployeesPatchObject(dequeuedTask.BusinessPartner.ContactEmployees, dequeuedTask.ExistentBusinessPartner.ContactEmployees);
+            //we don't want to update: CUITs/DNI and Currency
+            dequeuedTask.BusinessPartner.FederalTaxID = null;
+            dequeuedTask.BusinessPartner.Currency = null;
+
+            var sapResponse = await SendMessage(dequeuedTask);
+
+            var taskResult = new SapTaskResult
+            {
+                IsSuccessful = sapResponse.IsSuccessStatusCode,
+                SapResponseContent = await sapResponse.Content.ReadAsStringAsync(),
+                TaskName = "Updating Business Partner"
+            };
+
+            if (!taskResult.IsSuccessful && taskResult.SapResponseContent.Contains("[OCRD.FatherCard]"))
+            {
+                dequeuedTask.BusinessPartner.FatherCard = !dequeuedTask.BusinessPartner.CardCode.EndsWith(".0")
+                    ? dequeuedTask.BusinessPartner.CardCode.Replace(dequeuedTask.BusinessPartner.CardCode.Substring(dequeuedTask.BusinessPartner.CardCode.IndexOf(".")), ".0")
+                    : null;
+                return await RetryUpdateBusinessPartner(dequeuedTask);
+            }
```

Comment:
I think you don't need this. It was removed in the previous sap service.

Reply:
I took this from previous SAP service, this is the recursive call that I talked yesterday.  I will check this, maybe I saw an outdated version.

Reply:
This is correct. But in the BussinesPartnerService you are already checking for the father card code. So this is pointless now.
Maybe I missed something and didn't remove it completely in the previous service.

Reply:
As we talk, this was unnecesary so we removed. Thanks Fer!